### PR TITLE
Add extendedWkbGeometry to migrate parcel + import parcel

### PR DESCRIPTION
**Pull request by pgallik**

### CODE REVIEW

File: src/ParcelRegistry/Parcel/Events/ParcelSnapshotV2.cs [hunk 1]
```diff
@@ -30,8 +30,7 @@ public class ParcelSnapshotV2
             ParcelStatus parcelStatus,
             bool isRemoved,
             IEnumerable<AddressPersistentLocalId> addressPersistentLocalIds,
-            Coordinate? xCoordinate,
-            Coordinate? yCoordinate,
+            string extendedWkbGeometry,
```
Comment (ArneD) [inline src/ParcelRegistry/Parcel/Events/ParcelSnapshotV2.cs:33]: Shouldn't this be `ExtendedWkbGeometry`?

File: src/ParcelRegistry/Parcel/Events/ParcelSnapshotV2.cs [hunk 2]
```diff
@@ -6,7 +6,9 @@ namespace ParcelRegistry.Parcel.Events
     using Be.Vlaanderen.Basisregisters.AggregateSource.Snapshotting;
     using Be.Vlaanderen.Basisregisters.EventHandling;
     using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using NetTopologySuite.Geometries;
```
Comment (ArneD) [inline src/ParcelRegistry/Parcel/Events/ParcelSnapshotV2.cs:9]: usings?

File: src/ParcelRegistry.Projections.Legacy/ParcelDetailV2/ParcelDetailV2Projections.cs [hunk 3]
```diff
@@ -171,6 +172,23 @@ await context
                     },
                     ct);
             });
+
+            When<Envelope<ParcelWasImported>>(async (context, message, ct) =>
+            {
+                var item = new ParcelDetailV2(
```
Comment (ArneD) [inline src/ParcelRegistry.Projections.Legacy/ParcelDetailV2/ParcelDetailV2Projections.cs:178]: Need migration and geometry saved

File: src/ParcelRegistry.Projections.Legacy/ParcelSyndication/ParcelSyndicationProjections.cs [hunk 4]
```diff
@@ -125,18 +130,21 @@ await context
...
                 var parcelSyndicationItem = new ParcelSyndicationItem
                 {
                     Position = message.Position,
                     ParcelId = message.Message.ParcelId,
                     CaPaKey = message.Message.CaPaKey,
                     Status = ParcelStatus.Parse(message.Message.ParcelStatus),
-                    XCoordinate = message.Message.XCoordinate,
-                    YCoordinate = message.Message.YCoordinate,
+                    XCoordinate = decimal.Parse(geometry.CentroidWithinArea().Coordinate.X.ToString()),
+                    YCoordinate = decimal.Parse(geometry.CentroidWithinArea().Coordinate.Y.ToString()),
```
Comment (ArneD) [inline src/ParcelRegistry.Projections.Legacy/ParcelSyndication/ParcelSyndicationProjections.cs:147]: Create migration to remove XY coordinates and add geometry as extendedwkb (string = hex)

**Review by ArneD — CHANGES_REQUESTED** [review]

File: src/ParcelRegistry/Parcel/Commands/ImportParcel.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+        public Guid CreateCommandId()
+            => Deterministic.Create(Namespace, $"ImportParcel-{ToString()}");
+
+        public override string? ToString()
+            => ToStringBuilder.ToString(IdentityFields());
+
+        private IEnumerable<object> IdentityFields()
+        {
+            yield return VbrCaPaKey;
```
Comment (jvandaal) [inline src/ParcelRegistry/Parcel/Commands/ImportParcel.cs:40]: ParcelId is missing here.

File: src/ParcelRegistry/Parcel/Events/ParcelWasImported.cs [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+    using Be.Vlaanderen.Basisregisters.GrAr.Common;
+    using Be.Vlaanderen.Basisregisters.GrAr.Provenance;
+    using ParcelRegistry.Parcel;
+
+    [EventTags(EventTag.For.Sync, EventTag.For.Edit)]
+    [EventName(EventName)]
+    [EventDescription("Het perceel werd geïmporteerd.")]
+    public sealed class ParcelWasImported : IParcelEvent
+    {
+        public const string EventName = "ParcelGeometryWasImported"; // BE CAREFUL CHANGING THIS!!
```
Comment (jvandaal) [inline src/ParcelRegistry/Parcel/Events/ParcelWasImported.cs:17]: EventName doesn't match class name.

File: src/ParcelRegistry/Parcel/Parcel.cs [hunk 7]
```diff
@@ -18,9 +19,10 @@ public sealed partial class Parcel : AggregateRootEntity, ISnapshotable
             ParcelStatus parcelStatus,
             bool isRemoved,
             IEnumerable<AddressPersistentLocalId> addressPersistentLocalIds,
-            Coordinate? xCoordinate,
-            Coordinate? yCoordinate)
+            ExtendedWkbGeometry extendedWkbGeometry)
         {
+            GuardPolygon(WKBReaderFactory.Create().Read(extendedWkbGeometry));
```
Comment (jvandaal) [inline src/ParcelRegistry/Parcel/Parcel.cs:24]: I'm not 100% we want to perform this check when migrating? It seems logical we only want valid polygons in our domain, but just checking.

File: test/ParcelRegistry.Tests/AggregateTests/WhenImportingParcel/GivenParcelDoesNotExists.cs [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+    using Xunit;
+    using Xunit.Abstractions;
+
+    public class GivenParcelDoesNotExists : ParcelRegistryTest
+    {
+        public GivenParcelDoesNotExists (ITestOutputHelper testOutputHelper) : base(testOutputHelper)
+        {
+            Fixture.Customize(new WithFixedParcelId());
+            Fixture.Customize(new WithParcelStatus());
+            Fixture.Customize(new Legacy.AutoFixture.WithFixedParcelId());
```
Comment (jvandaal) [inline test/ParcelRegistry.Tests/AggregateTests/WhenImportingParcel/GivenParcelDoesNotExists.cs:26]: Is this last fixture necessary?

File: src/ParcelRegistry/Parcel/Parcel.cs (hunk 7 above)
Comment (pgallik) [reply src/ParcelRegistry/Parcel/Parcel.cs:24]: I would leave it, @ArneD ?

**Review by ArneD — APPROVED** [review]

**Pull request merged**